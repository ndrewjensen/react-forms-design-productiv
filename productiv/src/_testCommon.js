
const INITIAL_TODOS= [
  {
    id: 1,
    title: "Test1",
    description: "Test1 Description",
    priority: 1,
  },
  {
    id: 2,
    title: "Test2",
    description: "Test2 Description",
    priority: 2,
  },
     {
    id: 3,
    title: "Test3",
    description: "Test3 Description",
    priority: 3,
  }
];

export {INITIAL_TODOS}