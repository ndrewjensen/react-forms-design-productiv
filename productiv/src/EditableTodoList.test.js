import React from "react";
import { screen, render, fireEvent } from "@testing-library/react";

import { INITIAL_TODOS } from "./_testCommon"
import EditableTodoList from "./EditableTodoList";

const { getByDisplayValue } = screen;

describe("productiv EditableTodoList", function () {
  it("renders without crashing", function () {
    render(<EditableTodoList todos={INITIAL_TODOS} />);
  });

  it("renders EditableTodos", function () {
    const { container, debug } = render(<EditableTodoList todos={INITIAL_TODOS} />);
    expect(container.querySelectorAll(".EditableTodo").length).toEqual(3)
  });

  it("renders nothing if no todos", function () {
    const { container, debug } = render(<EditableTodoList todos={[]} />);
    expect(container.querySelectorAll(".EditableTodo").length).toEqual(0)
  });

  it("matches snapshot", function () {
    const { container } = render(<EditableTodoList todos={INITIAL_TODOS} />);
    expect(container).toMatchSnapshot();
  });
});
