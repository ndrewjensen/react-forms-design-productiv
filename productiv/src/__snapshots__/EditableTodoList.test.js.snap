// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`productiv EditableTodoList matches snapshot 1`] = `
<div>
  <div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
        >
          <div>
            <b>
              Test1
            </b>
            <small>
               priority: 
              1
            </small>
          </div>
          <div>
            <small>
              Test1 Description
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
        >
          <div>
            <b>
              Test2
            </b>
            <small>
               priority: 
              2
            </small>
          </div>
          <div>
            <small>
              Test2 Description
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
        >
          <div>
            <b>
              Test3
            </b>
            <small>
               priority: 
              3
            </small>
          </div>
          <div>
            <small>
              Test3 Description
            </small>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
